How to post video

post video seem not update

Hello,
While trying to upload a video, I get this:

FATAL EXCEPTION: AsyncTask #3
    Process: me.fponzi.instantsharing, PID: 16226
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:353)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.lang.NullPointerException
        at java.io.FileOutputStream.<init>(FileOutputStream.java:212)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:171)
        at dev.niekirk.com.instagram4android.requests.InstagramUploadVideoRequest.configureThumbnail(InstagramUploadVideoRequest.java:105)
        at dev.niekirk.com.instagram4android.requests.InstagramUploadVideoRequest.execute(InstagramUploadVideoRequest.java:84)
        at dev.niekirk.com.instagram4android.requests.InstagramUploadVideoRequest.execute(InstagramUploadVideoRequest.java:33)
        at dev.niekirk.com.instagram4android.Instagram4Android.sendRequest(Instagram4Android.java:241)
        at me.fponzi.instantsharing.utils.InstagramAPIWrapper$PostVideoAsyncTask.doInBackground(InstagramAPIWrapper.java:129)
        at me.fponzi.instantsharing.utils.InstagramAPIWrapper$PostVideoAsyncTask.doInBackground(InstagramAPIWrapper.java:108)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) 
        at java.lang.Thread.run(Thread.java:764) 

This is my code:
```
  public class PostVideoAsyncTask extends AsyncTask<String, Void, StatusResult>
    {
        public final String TAG = PostVideoAsyncTask.class.getCanonicalName();
        private final File video;
        private final ShareDoneCallback callback;

        public PostVideoAsyncTask(File video, ShareDoneCallback callback)
        {
            this.video = video;
            this.callback = callback;
        }
    
        @Override
        protected StatusResult doInBackground(String... strings)
        {
            try
            {

                Log.i(TAG, "doInBackground: " + (instagram == null));
                return instagram.sendRequest(new InstagramUploadVideoRequest( // line 129
                                    video,
                            Config.CAPTION)); 
    
            }
            catch (IOException e)
            {
                Log.e(TAG, "doInBackground: login error:", e );
                e.printStackTrace();
            }
            return null;
        }
    
        @Override
        protected void onPostExecute(StatusResult statusResult)
        {
            callback.sharedDoneCallback(statusResult.getStatus().equals("ok"));
        }
    }
```

Any suggestions? Thanks!

--- a/app/src/main/java/dev/niekirk/com/instagram4android/requests/internal/InstagramUploadVideoJobRequest.java
+++ b/app/src/main/java/dev/niekirk/com/instagram4android/requests/internal/InstagramUploadVideoJobRequest.java
@@ -18,109 +18,79 @@ import lombok.AllArgsConstructor;
 
 @AllArgsConstructor
 public class InstagramUploadVideoJobRequest extends InstagramRequest<StatusResult> {
-
+   private final String TAG = UploadVideoJobRequest.class.getCanonicalName();
     private String uploadId;
     private String uploadUrl;
     private String uploadJob;
     private File videoFile;
-
-
-    @Override
+    
     public String getUrl() {
-        return uploadUrl;
+        return this.uploadUrl;
     }
-
-    @Override
+    
     public String getMethod() {
         return "POST";
     }
-
-    @Override
-    public StatusResult execute() throws IOException {
-
+    
+    
+    private Request createHTTPRequest()
+    {
+        
+        return (new okhttp3.Request.Builder()).url(this.getUrl())
+                .addHeader("X-IG-Capabilities", "3Q4=")
+                .addHeader("X-IG-Connection-Type", "WIFI")
+                .addHeader("Cookie2", "$Version=1")
+                .addHeader("Accept-Language", "en-US")
+                .addHeader("Accept-Encoding", "gzip, deflate")
+                .addHeader("Session-ID", uploadId)
+                .addHeader("Connection", "keep-alive")
+                .addHeader("job", uploadJob)
+                .addHeader("Host", "upload.instagram.com")
+                .addHeader("User-Agent", InstagramConstants.USER_AGENT)
+                .post(createRequestBody())
+                .build();
+    }
+    
+    private RequestBody createRequestBody()
+    {
+        return new MultipartBody.Builder()
+                .setType(MultipartBody.FORM)
+                .addPart(Headers.of("Content-Disposition", "attachment; filename=\"video.mp4\""),
+                         RequestBody.create(MediaType.parse("video/mp4"), videoFile)
+                ).build();
+    }
+    
+    public StatusResult execute() throws IOException
+    {
+        
         String url = getUrl();
-        Log.d("UPLOAD", "URL Upload: " + url);
-
-        /*
-        HttpPost post = new HttpPost(url);
-        post.addHeader("X-IG-Capabilities", "3Q4=");
-        post.addHeader("X-IG-Connection-Type", "WIFI");
-        post.addHeader("Cookie2", "$Version=1");
-        post.addHeader("Accept-Language", "en-US");
-        post.addHeader("Accept-Encoding", "gzip, deflate");
-        post.addHeader("Content-Type", "application/octet-stream");
-        post.addHeader("Session-ID", uploadId);
-        post.addHeader("Connection", "keep-alive");
-        post.addHeader("Content-Disposition", "attachment; filename=\"video.mp4\"");
-        post.addHeader("job", uploadJob);
-        post.addHeader("Host", "upload.instagram.com");
-        post.addHeader("User-Agent", InstagramConstants.USER_AGENT);
-
-
-
-
-        log.info("User-Agent: " + InstagramConstants.USER_AGENT);
-
-        try (FileInputStream is = new FileInputStream(videoFile)) {
-            byte[] videoData = MyStreamUtils.readContentBytes(is);
-
-            //TODO: long ranges? need to handle?
-            int requestSize = (int) Math.floor(videoData.length / 4.0);
-            int lastRequestExtra = (int) (videoData.length - (requestSize * 3));
-
-
-            for (int i = 0; i < 4; i++) {
-
-                int start = i * requestSize;
-                int end;
-                if (i == 3) {
-                    end = i * requestSize + lastRequestExtra;
-                } else {
-                    end = (i + 1) * requestSize;
-                }
-
-                int actualLength = (i == 3 ? lastRequestExtra : requestSize);
-
-                String contentRange = String.format("bytes %s-%s/%s", start, end - 1, videoData.length);
-                //post.setHeader("Content-Length", String.valueOf(end - start));
-                post.setHeader("Content-Range", contentRange);
-
-                byte[] range = Arrays.copyOfRange(videoData, start, start + actualLength);
-                log.info("Total is " + videoData.length + ", sending " + actualLength + " (starting from " + start + ") -- " + range.length + " bytes.");
-
-                post.setEntity(EntityBuilder.create().setBinary(range).build());
-
-                try (CloseableHttpResponse response = api.getClient().execute(post)) {
-                    int resultCode = response.getStatusLine().getStatusCode();
-                    String content = EntityUtils.toString(response.getEntity());
-                    log.info("Result of part " + i + ": " + content);
-
-                    post.releaseConnection();
-                    response.close();
-
-                    if (resultCode != 200 && resultCode != 201) {
-                        throw new IllegalStateException("Failed uploading video (" + resultCode + "): " + content);
-                    }
-
-                }
-
-            }
-
-            return new StatusResult("ok");
+        Log.i(TAG, "Upload URL: " + url);
+        Response response = this.api.getClient().newCall(createHTTPRequest()).execute();
+        
+        this.api.setLastResponse(response);
+        int resultCode = response.code();
+        
+        String content = response.body().string();
+        Log.d(TAG, "MultiPart upload results: [" + resultCode + "] " + content);
+        
+        if (resultCode != 200 && resultCode != 201) {
+            throw new IllegalStateException("Failed uploading video (" + resultCode + "): " + content);
         }
-
-        */
         return new StatusResult("ok");
     }
-
-    @Override
+    
     public String getPayload() {
         return null;
     }
-
-    @Override
+    
     public StatusResult parseResult(int statusCode, String content) {
-        return parseJson(statusCode, content, StatusResult.class);
+        return (StatusResult)this.parseJson(statusCode, content, StatusResult.class);
     }
-
+    
+    public UploadVideoJobRequest(String uploadId, String uploadUrl, String uploadJob, File videoFile) {
+        this.uploadId = uploadId;
+        this.uploadUrl = uploadUrl;
+        this.uploadJob = uploadJob;
+        this.videoFile = videoFile;
+    }
 }

